src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py

```diff
@@ -2377,4 +2377,110 @@ def test_containerapp_max_inactive_revisions(self, resource_group):
+        self.cmd(f'containerapp show -g {resource_group} -n {app}', checks=[
+                JMESPathCheck("properties.configuration.runtime.java.enableMetrics", True)
+            ])
+
+        # Delete container app
+        self.cmd(f'containerapp delete  -g {resource_group} -n {app} --yes')
+
+        # Create container app with disabling java metrics
+        self.cmd(f'containerapp create -g {resource_group} -n {app} --image {image} --environment {env} --enable-java-metrics=false')
+        self.cmd(f'containerapp show -g {resource_group} -n {app}', checks=[
+                JMESPathCheck("properties.configuration.runtime.java.enableMetrics", False)
+            ])
+
+        # Delete container app
+        self.cmd(f'containerapp delete  -g {resource_group} -n {app} --yes')
+
+        # Create container app failed with wrong runtime
+        self.cmd(f'containerapp create -g {resource_group} -n {app} --image {image} --environment {env} --runtime=generic --enable-java-metrics',
+                 expect_failure=True)
+
+        # Create container app failed with unsupported runtime
+        self.cmd(f'containerapp create -g {resource_group} -n {app} --image {image} --environment {env} --runtime=otherlanguage',
+                 expect_failure=True)
+
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="northcentralus")
+    def test_containerapp_runtime_java_metrics_update(self, resource_group):
+        TEST_LOCATION = "northcentralusstage"
+        self.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+
+        app = self.create_random_name(prefix='aca', length=24)
+        image = "mcr.microsoft.com/k8se/quickstart:latest"
+
+        env = prepare_containerapp_env_for_app_e2e_tests(self)
+
+        # Create container app without enabling java metrics
+        self.cmd(f'containerapp create -g {resource_group} -n {app} --image {image} --environment {env}')
+        self.cmd(f'containerapp show -g {resource_group} -n {app}', checks=[
+                JMESPathCheck("properties.configuration.runtime", None)
+            ])
+
+        # Update contaier app with default settings
+        self.cmd(f'containerapp update -g {resource_group} -n {app} --runtime=java', checks=[
+                JMESPathCheck("properties.configuration.runtime.java.enableMetrics", False)
+            ])
+
+        # Update container app to enable java metrics
+        self.cmd(f'containerapp update -g {resource_group} -n {app} --enable-java-metrics', checks=[
+                JMESPathCheck("properties.configuration.runtime.java.enableMetrics", True)
+            ])
+
+        # Update container app to disable java metrics
+        self.cmd(f'containerapp update -g {resource_group} -n {app} --enable-java-metrics=false', checks=[
+                JMESPathCheck("properties.configuration.runtime.java.enableMetrics", False)
```

Comment:
Please check if the `provisioningState` is `succeed` for every create/update command.